src/interpreter.c

```diff
@@ -23,9 +23,38 @@ typedef struct {
     size_t ip; // Leak the currently-evaluating statement index to backtrace capture
     int preevaluation; // use special rules for pre-evaluating expressions (deprecated--only for ccall handling)
     int continue_at; // statement index to jump to after leaving exception handler (0 if none)
+    void *associated_frame_pointer;
+    jl_gcframe_t frame;
 } interpreter_state;
 
-#include "interpreter-stacktrace.c"
+#ifdef __clang_analyzer__
+
+extern void JL_GC_ENABLEFRAME(interpreter_state*) JL_NOTSAFEPOINT;
+
+// This is necessary, because otherwise the analyzer considers this undefined
+// behavior and terminates the exploration
+#define JL_GC_PUSHFRAME(frame, n)     \
+  frame = (interpreter_state*)alloca(sizeof(*frame) + sizeof(void*) * (n)); \
+  memset(&frame[1], 0, sizeof(void*) * n); \
+  _JL_GC_PUSHARGS((jl_value_t**)&frame[1], n);
+
+#else
+
+#define JL_GC_PUSHFRAME(frame,n)                                                    \
```

Comment:
This name makes me think the macro mainly to do with GC. But it's also very much to do with the interpreter. I don't have a great suggestion for a better name though... `JL_GC_PUSH_INTERPFRAME`?

Reply:
The macro only does things with `gc`. It doesn't care what else `frame` happens to be or to do.

Reply:
Well it sets aside the space and names the type `(interpreter_state*)`. A normal JL_GC_PUSH doesn't do those things.